reactive/kubernetes_master.py

```diff
@@ -1202,35 +1156,36 @@ def push_service_data():
         kube_api.configure(kubernetes_master.STANDARD_API_PORT)
 
 
-@when('certificates.available', 'kube-api-endpoint.available')
+@when('certificates.available', 'kube-api-endpoint.available', 'cni.available')
 def send_data():
     '''Send the data that is required to create a server certificate for
     this server.'''
-    kube_api_endpoint = endpoint_from_flag('kube-api-endpoint.available')
-
     # Use the public ip of this unit as the Common Name for the certificate.
     common_name = hookenv.unit_public_ip()
 
-    # Get the SDN gateway based on the cidr address.
-    kubernetes_service_ip = get_kubernetes_service_ip()
+    # Get the SDN gateways based on the service CIDRs.
+    k8s_service_ips = kubernetes_master.get_kubernetes_service_ips()
 
-    # Get ingress address
-    ingress_ip = get_ingress_address(kube_api_endpoint.endpoint_name)
+    cluster_cidr = kubernetes_common.cluster_cidr()
+    bind_ips = kubernetes_common.get_bind_addrs(
+        ipv4=kubernetes_common.is_ipv4(cluster_cidr),
+        ipv6=kubernetes_common.is_ipv6(cluster_cidr),
+    )
 
     domain = hookenv.config('dns_domain')
     # Create SANs that the tls layer will add to the server cert.
     sans = [
-        hookenv.unit_public_ip(),
-        ingress_ip,
```

Comment:
Why is it safe to remove ingress_ip? I would expect it to still be needed - it's the IP that clients of the kube-api-endpoint relation use to connect to the API.

Reply:
It should be covered by `get_bind_addrs()` but I'll add it back explicitly just in case it isn't.